dev/com.ibm.ws.kernel.service/src/com/ibm/ws/common/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,382 @@
+    }
+
+    public static MessageDigest getMessageDigest(String algorithm) throws NoSuchAlgorithmException {
+        if (!supportedMessageDigestAlgorithms.contains(algorithm))
+            throw new NoSuchAlgorithmException(String.format("Algorithm %s is not supported", algorithm));
+        return MessageDigest.getInstance(algorithm);
+    }
+
+    public static MessageDigest getMessageDigestForLTPA() {
+        MessageDigest md1 = null;
+        try {
+            if (fipsEnabled) {
+                if (isSemeruFips()) {
+                    md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA256,
+                                                    OPENJCE_PLUS_FIPS_NAME);
+                } else {
+                    md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA256,
+                                                    IBMJCE_PLUS_FIPS_NAME);
+                }
+            } else {
+                md1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM_SHA);
+            }
+
+        } catch (NoSuchAlgorithmException e) {
+            // instrumented ffdc
+        } catch (NoSuchProviderException e) {
+            // instrumented ffdc;
+        }
+
+        return md1;
+    }
+
+    public static String getProperty(final String prop, final String defaultValue) {
+        return AccessController.doPrivileged(new PrivilegedAction<String>() {
+            @Override
+            public String run() {
+                return System.getProperty(prop, defaultValue).toLowerCase();
+            }
+        });
+    }
+
+    static String getFipsLevel() {
+        return getProperty("com.ibm.fips.mode", "disabled");
+    }
+
+    public static boolean isSemeruFips() {
+        return "true".equals(getProperty("semeru.fips", "false"));
+    }
+
+    public static boolean isFips140_3Enabled() {
+
+        return isRunningBetaMode() &&
+               "140-3".equals(FIPSLevel) || "true".equalsIgnoreCase(getProperty("global.fips_140-3", "false")) || isSemeruFips();
+    }
+
+    public static boolean isFips140_2Enabled() {
+        return isRunningBetaMode() && "140-2".equals(FIPSLevel);
+    }
+
+    public static boolean isFIPSEnabled() {
+        if (fipsChecked) {
+            return fipsEnabled;
+        } else {
+            fipsEnabled = isFips140_2Enabled() || isFips140_3Enabled();
+            fipsChecked = true;
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "isFIPSEnabled: " + fipsEnabled);
+            }
+            return fipsEnabled;
+        }
+    }
+
+    public static boolean isRunningBetaMode() {
+
+        if (!ProductInfo.getBetaEdition()) {
+            return false;
+        } else {
+            // Running beta exception, issue message if we haven't already issued one for this class
+            if (!issuedBetaMessage) {
+                Tr.info(tc, "BETA: A beta method has been invoked for the class CryptoUtils for the first time.");
```

Comment:
Today this message will come out even if the user has not configured 140-3 for example in `isFips140_3Enabled()` because the conditional is backwards in there.  It check to see if beta it set before checking to see if 140-3 is set.  No reason to put out a message unless they are using function that requires beta.

Reply:
Fixed it.